contracts/upgradeable_contracts/erc20_to_erc20/FeeManagerErcToErcPOSDAO.sol

```diff
@@ -0,0 +1,36 @@
+pragma solidity 0.4.24;
+
+import "../BaseFeeManager.sol";
+import "../../IBlockReward.sol";
+
+contract FeeManagerErcToErcPOSDAO is BaseFeeManager {
+
+    function getFeeManagerMode() public pure returns(bytes4) {
+        return bytes4(keccak256(abi.encodePacked("manages-both-directions")));
+    }
+
+    function blockRewardContract() internal view returns(IBlockReward) {
+        return IBlockReward(addressStorage[keccak256(abi.encodePacked("blockRewardContract"))]);
+    }
+
+    function distributeFeeFromAffirmation(uint256 _fee) external {
+        distributeFeeFromBlockReward(_fee);
+    }
+
+    function distributeFeeFromSignatures(uint256 _fee) external {
+        distributeFeeFromBlockReward(_fee);
+    }
+
+    function distributeFeeFromBlockReward(uint256 _fee) internal {
+        IBlockReward blockReward = blockRewardContract();
+        blockReward.addBridgeTokenFeeReceivers(_fee);
+    }
+
+    function isPOSDAOFeeManager() public pure returns(bool) {
+        return true;
+    }
+
+    function onAffirmationFeeDistribution(address _rewardAddress, uint256 _fee) internal {}
```

Comment:
Should we still have these declarations? they are internal so should not be reflected in the ABI of the contract.

Reply:
Those declarations were needed because they were abstract methods defined on `BaseFeeManager`. I made a refactor for this here https://github.com/poanetwork/poa-bridge-contracts/pull/166/commits/cf81b767ba72e4ce120b3bd536133fc5a7a0ab7c.
I left basic methods on `BaseFeeManager` and added two new contracts that extend it: 
- `ValidatorsFeeManager`:  Includes the logic to distribute fee proportionally using validators contract.
- `BlockRewardFeeManager`: Includes the logic to delegate fee distribution to block reward contract.

Reply:
It looks very smart. Cool!